src/OpenTelemetry/Internal/Options/ConfigurationExtensions.cs

```diff
@@ -146,7 +146,7 @@ internal static class ConfigurationExtensions
         Debug.Assert(services != null, "services was null");
         Debug.Assert(optionsFactoryFunc != null, "optionsFactoryFunc was null");
 
-        services!.TryAddSingleton<IOptionsFactory<T>>(sp =>
+        services.TryAddSingleton<IOptionsFactory<T>>(sp =>
```

Comment:
Why remove the null-forgiving operator?

Reply:
Because there is a `Debug.Assert(services != null` right above it. So it is clutter and unnecessary. The compiler knows that `services` is not null here.

However, for some reason on line 159 it still requires the `!` for the netstandard builds. I'm assuming that is a bug in the compiler somehow? I left that one since it doesn't build without it (and thus is necessary).